src/test/java/nl/novi/serviceconnect/services/TransactionServiceTest.java

```diff
@@ -0,0 +1,230 @@
+package nl.novi.serviceconnect.services;
+
+import nl.novi.serviceconnect.dtos.TransactionInputDto;
+import nl.novi.serviceconnect.dtos.TransactionUpdateDto;
+import nl.novi.serviceconnect.dtos.TransactionOutputDto;
+import nl.novi.serviceconnect.exceptions.BadRequestException;
+import nl.novi.serviceconnect.exceptions.RecordNotFoundException;
+import nl.novi.serviceconnect.models.ServiceRequest;
+import nl.novi.serviceconnect.models.Transaction;
+import nl.novi.serviceconnect.repository.ServiceRequestRepository;
+import nl.novi.serviceconnect.repository.TransactionRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.ResponseEntity;
+
+import java.io.File;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+import static org.mockito.Mockito.never;
+
+@ExtendWith(MockitoExtension.class)
+class TransactionServiceTest {
+
+    @Mock
+    TransactionRepository transactionRepository;
+
+    @Mock
+    ServiceRequestRepository serviceRequestRepository;
+
+    @InjectMocks
+    TransactionService transactionService;
+
+    @Test
+    public void shouldGetServiceRequest() {
+
+        //Arrange
+        ServiceRequest mockServiceRequest = new ServiceRequest();
+        mockServiceRequest.setId(1L);
+
+        //Act
+        when(serviceRequestRepository.findById(1L)).thenReturn(Optional.of(mockServiceRequest));
+        ServiceRequest result = transactionService.getServiceRequest(1L);
+
+        //Assert
+        assertNotNull(result);
+        assertEquals(1L, result.getId());
+    }
+
+    @Test
+    public void shouldSaveFile_Success() {
+
+        //Arrange
+        String fileSeparator = File.separator;
+
+        //Act
+        String filePath = transactionService.saveFile();
+        String expectedFilePathPrefix = "invoicesPdf" + fileSeparator + "invoice_";
+
+        //Assert
+        assertTrue(filePath.startsWith(expectedFilePathPrefix), "File path should start with: " + expectedFilePathPrefix);
+    }
+
+    @Test
+    public void shouldGetServiceRequestNonExistingId() {
```

Comment:
Ik kon uit de naam niet halen wat het deed. Ik dacht dat het zou kijken naar nog niet gereserveerde ids of iets dergelijks

Reply:
Changed to shouldThrowException_NoServiceRequestId()